lib/profile.dart

```diff
@@ -1,23 +1,145 @@
 import 'package:flutter/material.dart';
 
+import 'appstate.dart';
+import 'strings.dart';
+
+// ignore: must_be_immutable
 class UserProfilePage extends StatefulWidget {
-  final String title = 'Profile';
+  AppState appState;
 
-  const UserProfilePage({Key? key}) : super(key: key);
+  UserProfilePage({Key? key, required this.appState}) : super(key: key);
 
   @override
   State<UserProfilePage> createState() => _UserProfilePageState();
 }
 
 class _UserProfilePageState extends State<UserProfilePage> {
+  final _formKey = GlobalKey<FormState>();
+  final _statusMessageInputController = TextEditingController();
+  final _nickInputController = TextEditingController();
+  bool _applyButtonPressed = false;
+
+  void _onUpdateNick() {
+    if (_formKey.currentState!.validate()) {
+      widget.appState.setNick(_nickInputController.text);
+    }
+  }
+
+  void _onUpdateStatusMessage() {
+    if (_formKey.currentState!.validate()) {
+      widget.appState.setUserStatus(_statusMessageInputController.text);
+    }
+  }
+
+  bool _nickIsValid(String nick) {
+    return _nickInputController.text.isNotEmpty &&
+        _nickInputController.text.length <= 32;
+  }
+
+  bool _statusMessageIsValid(String nick) {
+    return _statusMessageInputController.text.length <= 256;
+  }
+
+  _onValidate() {
+    if (!_nickIsValid(_nickInputController.text) ||
+        !_statusMessageIsValid(_statusMessageInputController.text)) {
+      return;
+    }
+
+    if (widget.appState.getNick() != _nickInputController.text ||
+        widget.appState.getUserStatus() != _statusMessageInputController.text) {
+      _setApplyButtonState(true);
+      _onUpdateNick();
+      _onUpdateStatusMessage();
+      Navigator.pop(context);
+    }
+  }
+
+  void _setApplyButtonState(bool pressed) {
+    setState(() {
+      _applyButtonPressed = pressed;
+    });
+  }
+
   @override
   Widget build(BuildContext context) {
+    _nickInputController.text = widget.appState.getNick();
+    _statusMessageInputController.text = widget.appState.getUserStatus();
+
     return Scaffold(
       appBar: AppBar(
-        title: Text(widget.title),
+        title: const Text(Strings.menuProfile),
       ),
-      body: const Center(
-        child: Text('Username'),
+      body: Form(
+        key: _formKey,
+        child: Column(
+          children: [
+            const Padding(
+              padding: EdgeInsets.all(10),
```

Comment:
I think in general, all UI distances in Flutter are powers of two. I know e.g. the ListTile uses 16 as its default padding.